exploration/syntax-exploration-2.md

```diff
@@ -0,0 +1,257 @@
+| 3a     | Use sigils for code mode                                       | +               | -                         | +                              | +                | + |
+
+## Candidates
+
+### Candidate 1a
+
+**Description:** Invert for Text Mode, distinguish statements from placeholders
+
+- Placeholders and expressions use `{`/`}`:
+  > `{$var}`, `{unquoted}`, `{|quoted|}`, `{$var :function}`
+- Non-placeholders use `{#`/`}`:
+  > `{#input $user}`, `{#local $var=$foo :expr}`, `{#match {$foo}}`, `{#when * *}`
+
+The use of `#` as the sigil is intentional here.
+This sigil might be replaced, but does not require additional escaping, as code is enclosed in `{`/`}`.
+Note that the `#` sigil might also be considered to be part of the keyword, e.g. `#when`, `#local` vs.
+allowing whitespace to appear between them.
+
+Patterns in this syntax are generally unquoted,
+but MAY be quoted with the 2023-10-09 consensus `{{`/`}}` quotes.
+
+ASCII whitespace trimming is assumed, but is a hot topic.
+
+**Examples**
+```
+Hello world!
+```
+```
+Hello {$user}
+```
+```
+{#input $user :person type=informal}
+Hello {$user}
+
+{#input $user :person type=informal}Hello {$user}
+```
+```
+{#input $var :function option=value}
+{#local $foo = $bar :function option=value}
+Hello {$var}, you have a {$foo}
+
+{#input $var :function option=value}{#local $foo = $bar :function option=value}Hello {$var}, you have a {$foo}
+```
+```
+{#match {$foo}}
+{#when foo} Hello {$foo} you have a {$var}
+{#when *} {$foo} hello you have a {$var}
+
+{#match {$foo}}{#when foo}Hello {$foo} you have a {$var}{#when *}{$foo} hello you have a {$var}
+```
+```
+{#match {$foo :function option=value} {$bar :function option=value}}
+{#when a b} {{  {$foo} is {$bar}  }}
+{#when x y} {{  {$foo} is {$bar}  }}
+{#when * *} {|  |}{$foo} is {$bar}{|  |}
+
+{#match {$foo :function option=value}{$bar :function option=value}}{#when a b}{{  {$foo} is {$bar}  }}{#when x y}{{  {$foo} is {$bar}  }}{#when * *}{|  |}{$foo} is {$bar}{|  |}
+```
+
+### Candidate 2a
+
+**Description**: Text First, Current Syntax for "Complex" Messages
+
+Starts in text mode, but switches to the current "code-mode" syntax for any message containing declarations or selectors.
+
+- "Simple" messages (lacking declarations or selectors) are unquoted patterns.
+- Patterns with declarations (`input`, `local`), selectors (`match`/`when`) or both quote all patterns.
+- Pattern whitespace is always significant.
+  Note that "pattern whitespace" is only that whitespace that appears inside the quoting `{`/`}`
+  for quoted patterns.
+
+> The use in the previous iteration of `{{` and `}}` surrounding code-mode messages was identified as a potential negative, 
+> as the closing `}}` are superfluous to the message.
+
+The use of `>>` to represent the "starting code-mode sigil" is **_not_** final.
+**_Do not fixate on the specific character sequence when choosing (or not) this design._**
```

Comment:
```suggestion
The use of `{# ... #}` to represent the starting and ending code-mode sigils is **_not_** final.
**_Do not fixate on the specific character sequence when choosing (or not) this design._**
```

@mihnita or anyone else: Thoughts? I'm not particular about which delimiter, but using double curlies was too curly and too many for folks. Something to break up the visual monotony would be good, hence the `#` on the inside.

FYI, whatever we decide here should be reflected in the examples below, too.

Reply:
I actually like the opening delimiter only solution. I think it adds more clarity and less complexity:
- No need to balance the delimiter 'at a distance'
- You could only use it at the very start of a message, there should be nothing before this delimiter, except maybe insignificant white space.
- After entering code-mode, you can not get out of it. The closing delimiter thus would be somewhat redundant.